RANGER-5310:Include Apache Tez as the process framework for ranger-hive docker

rameeshm opened the pull request:
## What changes were proposed in this pull request?

Include Apache Tez as the process framework for ranger-hive docker 

- This fixes issue with Insert command in beeline
- Data processing is much faster with Tez's  DAG for processing.


## How was this patch tested?
Testing in Docker running  HiveServer 2 beeline and execute "INSERT" statement for DAG.
<img width="1541" height="611" alt="TezJob" src="https://github.com/user-attachments/assets/723fb01e-023f-4b4d-90b2-c0f7cdba85ec" />

rameeshm commented on dev-support/ranger-docker/scripts/hive-site-oracle.xml:
```diff
@@ -49,4 +49,72 @@
+    <property>
+        <name>tez.queue.name</name>
+        <value>default</value>
+        <description>YARN configuration for Tez jobs</description>
```
This just a naming of the description which can be ignored.

mneethiraj commented on dev-support/ranger-docker/scripts/create-users.sh:
```diff
@@ -0,0 +1,62 @@
+    echo "Created hadoop group"
+fi
+
+# Create alice user (uid: 2001, gid: 1001 - hadoop group)
```
Is it necessary for users `alice` and `abram` to be in `hadoop` group? This group is meant for service accounts (like `hive`, `hbase`, `impala`) to get certain privileges. User accounts shouldn't need to be part of this group.

Also, it is mandatory that uid for the created users be 2001 and 2002? If not, I suggest creating the users without specifying uid.

rameeshm replied:
Thanks @mneethiraj  for the review. Addressed this

rameeshm commented on dev-support/ranger-docker/.env:
```diff
@@ -25,8 +25,8 @@ SOLR_VERSION=8.11.3
 HBASE_VERSION=2.6.0
 HIVE_VERSION=4.0.1
-HIVE_HADOOP_VERSION=3.1.1
-KAFKA_VERSION=2.8.2
```
Looks like I accidently removed Kafka Version so need fix this.

rameeshm commented on dev-support/ranger-docker/scripts/create-users.sh:
```diff
@@ -0,0 +1,62 @@
+    fi
+}
+
+# Ensure hadoop group exists (gid 1001 is used by hdfs, yarn, hive users)
```
Addressed this issue as well

kumaab commented:
Thank you @rameeshm for the patch, I believe this is tested with Ubuntu base image, please see if this can be tested with UBI base image as well, this change needs to made in `.env` file: RANGER_BASE_VERSION=[20250712-1-ubi-8](https://hub.docker.com/layers/apache/ranger-base/20250712-1-ubi-8/images/sha256-f2bd4fe9d5e3a4e5ca363c61fb2a46bdb718dd208c7356fb654fc640717a189f), thanks!

kumaab commented on dev-support/ranger-docker/scripts/ranger-hive.sh:
```diff
@@ -50,14 +91,49 @@ fi
+fi
+
+# Additional verification: Check if HiveServer2 is listening on port 10000
+echo "Checking if HiveServer2 is listening on port 10000..."
```
health checks for HiveServer2 can be added in `docker-compose.ranger-hive.yml` as well, this PR adds healthchecks for all containers: #604

rameeshm commented:
> Thank you @rameeshm for the patch, I believe this is tested with Ubuntu base image, please see if this can be tested with UBI base image as well, this change needs to made in `.env` file: RANGER_BASE_VERSION=[20250712-1-ubi-8](https://hub.docker.com/layers/apache/ranger-base/20250712-1-ubi-8/images/sha256-f2bd4fe9d5e3a4e5ca363c61fb2a46bdb718dd208c7356fb654fc640717a189f), thanks!

@kumaab current patch with the review comments tested with RANGER_BASE_VERSION=[20250712-1-ubi-8]

rameeshm commented on dev-support/ranger-docker/scripts/create-users.sh:
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
```
@mneethiraj  Addressed this review comment to use ranger-base image and create users. Thanks

mneethiraj commented on dev-support/ranger-docker/Dockerfile.ranger-usersync:
```diff
@@ -14,9 +14,7 @@
-ARG RANGER_BASE_IMAGE
-ARG RANGER_BASE_VERSION
-FROM ${RANGER_BASE_IMAGE}:${RANGER_BASE_VERSION}
+FROM ranger
```
Why is it necessary to change the base image (`FROM @{RANGER_BASE_IMAGE}`) to `ranger`?

`ranger` is the image containing Ranger admin server; usersync shouldn't use `ranger` as the base.

rameeshm commented on dev-support/ranger-docker/scripts/hive-site-sqlserver.xml:
```diff
@@ -47,4 +47,72 @@
         <name>hive.zookeeper.client.port</name>
         <value>2181</value>
     </property>
+
```
@kumaab Lets address this in the JIRA which deprecates SQL Server

rameeshm commented:
> > @kumaab Addressed the review comments. Thanks
> 
> The docker build seems to be failing, please retain `OZONE_RUNNER_VERSION=20230615-1` in `.env`. thanks.

@kumaab addressed this issue. Thanks

kumaab reviewed (approved):
Looks good, tested the changes on my local as well as the docker build was failing in CI. thank you!
